core/cont/inc/TArrayC.h

```diff
@@ -42,6 +42,8 @@ class TArrayC : public TArray {
    void          Copy(TArrayC &array) const {array.Set(fN,fArray);}
    const Char_t *GetArray() const { return fArray; }
    Char_t       *GetArray() { return fArray; }
+   const void *GetVoidArray() const { return static_cast<const void *>(fArray); }
+   void *GetVoidArray() { return static_cast<void *>(fArray); }
```

Comment:
If we can get away with it, I think it would be better to have only the `const` overload. Also the name `GetDataPtr` may or may not be a clearer name.